collector/src/main/java/com/bitmovin/analytics/retryBackend/RetryQueue.kt

```diff
@@ -0,0 +1,79 @@
+package com.bitmovin.analytics.retryBackend
+
+import android.util.Log
+import com.bitmovin.analytics.utils.Util
+import java.lang.Exception
+import java.util.Calendar
+import java.util.Date
+import java.util.concurrent.locks.ReentrantLock
+import kotlin.Comparator
+import kotlin.math.pow
+
+class RetryQueue {
+    private val TAG = "RetryQueue"
+    private val lock = ReentrantLock()
+    private var retrySamplesSet = mutableListOf<RetrySample<Any>>()
```

Comment:
why is this called set if it's a list?
wouldn't that work easier (so you wouldn't have to sort every time after adding a new sample)? 
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/sorted-set-of.html

Reply:
Tried to implement set with comparator and it's really hard and messy to be able to add `retrySamples` with diffrent scheduled time and still have a equality check for event object which is generic.... All the time I got either samples being discarded because of same scheduled time or if I managed to add them into the set then `remove` method stops working because it can't determine which is correct sample to remove (and this causes literary hundreds of same samples being sent to backend in a few seconds in a infinite loop). Things like PriorityQueue would require upgrade of API verson :(
So simplest and most functional and safe solution is to use lists